### PR TITLE
BUG: date_range doesn't propagate ambigous=False to tz_localize

### LINKED ISSUE #35297: BUG: AmbiguousTimeError when using date_range even when argument ambiguous is set
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
>>> import pandas as pd
>>> timezone = 'America/New_York'
>>> start = pd.Timestamp(year=2020, month=11, day=1, hour=1).tz_localize(timezone, ambiguous=False)
>>> pd.date_range(start, periods=2, ambiguous=False)

---------------------------------------------------------------------------
AmbiguousTimeError                        Traceback (most recent call last)
<ipython-input-5-d142fd70e406> in <module>
      2 timezone = 'America/New_York'
      3 start = pd.Timestamp(year=2020, month=11, day=1, hour=1).tz_localize(timezone, ambiguous=False)
----> 4 pd.date_range(start, periods=2, ambiguous=False)

~/anaconda3/envs/oids_tst/lib/python3.6/site-packages/pandas/core/indexes/datetimes.py in date_range(start, end, periods, freq, tz, normalize, name, closed, **kwargs)
   1178         normalize=normalize,
   1179         closed=closed,
-> 1180         **kwargs,
   1181     )
   1182     return DatetimeIndex._simple_new(dtarr, tz=dtarr.tz, freq=dtarr.freq, name=name)

~/anaconda3/envs/oids_tst/lib/python3.6/site-packages/pandas/core/arrays/datetimes.py in _generate_range(cls, start, end, periods, freq, tz, normalize, ambiguous, nonexistent, closed)
    419                 # start/end as well to compare
    420                 if start is not None:
--> 421                     start = start.tz_localize(tz).asm8
    422                 if end is not None:
    423                     end = end.tz_localize(tz).asm8

pandas/_libs/tslibs/timestamps.pyx in pandas._libs.tslibs.timestamps.Timestamp.tz_localize()

pandas/_libs/tslibs/tzconversion.pyx in pandas._libs.tslibs.tzconversion.tz_localize_to_utc()

AmbiguousTimeError: Cannot infer dst time from 2020-11-01 01:00:00, try using the 'ambiguous' argument
```

#### Problem description

When the start time is at the day light saving ambiguous time, date_range raises errors even when the argument 'ambiguous' is set. It should not be raising error when the argument is set.

Issue could be due to the line
https://github.com/pandas-dev/pandas/blob/32ed15d39312193a8c36171f0e120d0f46a26c2c/pandas/core/arrays/datetimes.py#L421
does not correctly pass the argument ambiguous to the function `tz_localize()`.

#### Expected Output

No error

#### Output of ``pd.show_versions()``

AmbiguousTimeError: Cannot infer dst time from 2020-11-01 01:00:00, try using the 'ambiguous' argument

### PR DESCRIPTION
**Pull request by arw2019**
- [x] closes #35297 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -418,9 +418,9 @@ def _generate_range(
                 # index is localized datetime64 array -> have to convert
                 # start/end as well to compare
                 if start is not None:
-                    start = start.tz_localize(tz).asm8
+                    start = start.tz_localize(tz, ambiguous).asm8
```
Comment (mroeschke) [inline pandas/core/arrays/datetimes.py:421]: Could you also pass `nonexistent` and write a test case for that issue as well?

File: pandas/tests/indexes/datetimes/test_constructors.py [hunk 2]
```diff
@@ -787,6 +787,23 @@ def test_construction_with_nat_and_tzlocal(self):
         expected = DatetimeIndex([Timestamp("2018", tz=tz), pd.NaT])
         tm.assert_index_equal(result, expected)
 
+    def test_constructor_with_ambiguous_keyword_arg(self):
+        # GH 35297
+
+        timezone = "America/New_York"
+        start = pd.Timestamp(year=2020, month=11, day=1, hour=1).tz_localize(
```
Comment (mroeschke) [inline pandas/tests/indexes/datetimes/test_constructors.py:794]: Could you also test where the end is `ambiguous`

File: doc/source/whatsnew/v1.1.0.rst [hunk 3]
```diff
@@ -941,7 +941,7 @@ Timezones
 ^^^^^^^^^
 
 - Bug in :func:`to_datetime` with ``infer_datetime_format=True`` where timezone names (e.g. ``UTC``) would not be parsed correctly (:issue:`33133`)
--
+- Bug in :func:`date_range` was raising AmbiguousTimeError for valid input with `ambiguous=False` (:issue:`35297`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.1.0.rst:944]: Could you move this to 1.2.0?

File: pandas/tests/indexes/datetimes/test_constructors.py (hunk 2 above)
Comment (arw2019) [reply pandas/tests/indexes/datetimes/test_constructors.py:802]: added

Comment (arw2019) [reply pandas/tests/indexes/datetimes/test_constructors.py:802]: added

File: pandas/core/arrays/datetimes.py (hunk 1 above)
Comment (arw2019) [reply pandas/core/arrays/datetimes.py]: added both

File: doc/source/whatsnew/v1.1.0.rst (hunk 3 above)
Comment (arw2019) [reply doc/source/whatsnew/v1.1.0.rst:944]: For sure! I don't think we have `doc/source/whatsnew/v1.2.0.rst` atm - shall I create it?

Comment (mroeschke) [reply doc/source/whatsnew/v1.1.0.rst:944]: Sure that'd be great.

**Comment by arw2019** [comment]
@mroeschke Getting an error in the doctests that I think is unrelated to this PR:
```
=================================== FAILURES ===================================
__________________ [doctest] pandas.core.frame.DataFrame.info __________________
2593             Data columns (total 3 columns):
2594              #   Column    Non-Null Count    Dtype
2595             ---  ------    --------------    -----
2596              0   column_1  1000000 non-null  object
2597              1   column_2  1000000 non-null  object
2598              2   column_3  1000000 non-null  object
2599             dtypes: object(3)
2600             memory usage: 22.9+ MB
2601 
2602             >>> df.info(memory_usage='deep')
Differences (unified diff with -expected +actual):
    @@ -2,9 +2,9 @@
     RangeIndex: 1000000 entries, 0 to 999999
     Data columns (total 3 columns):
    - #   Column    Non-Null Count    Dtype
    ----  ------    --------------    -----
    + #   Column    Non-Null Count    Dtype 
    +---  ------    --------------    ----- 
      0   column_1  1000000 non-null  object
      1   column_2  1000000 non-null  object
      2   column_3  1000000 non-null  object
     dtypes: object(3)
    -memory usage: 188.8 MB
    +memory usage: 165.9 MB
```
I merged with master just now so I don't think it's because this branch is out-of-date

**Comment by mroeschke** [comment]
No worries, looks unrelated to this PR

Actually sorry, could you move the creation of the 1.2.0 whatsnew doc to a separate PR? Would make this PR cleaner.

File: doc/source/whatsnew/v1.1.0.rst [hunk 4]
```diff
@@ -941,7 +941,6 @@ Timezones
 ^^^^^^^^^
 
 - Bug in :func:`to_datetime` with ``infer_datetime_format=True`` where timezone names (e.g. ``UTC``) would not be parsed correctly (:issue:`33133`)
--
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.1.0.rst:944]: This change can be undone now.

File: doc/source/whatsnew/v1.2.0.rst [hunk 5]
```diff
@@ -0,0 +1,210 @@
+.. _whatsnew_120:
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.2.0.rst:1]: Same with this one (since you opened up the other PR with the new whatsnew file)

**Review by mroeschke — APPROVED** [review]

**Comment by simonjayhawkins** [comment]
Thanks @arw2019

**Pull request merged**